debugutils/aggregator.go

```diff
@@ -0,0 +1,109 @@
+package debugutils
+
+import (
+	"io"
+	"path/filepath"
+
+	"github.com/solo-io/go-utils/errors"
+	"github.com/solo-io/go-utils/installutils/helmchart"
+	"github.com/solo-io/go-utils/tarutils"
+	"github.com/spf13/afero"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	aggregatorName = "aggregator"
+)
+
+type Aggregator struct {
+	resourceCollector ResourceCollector
+	logCollector      LogCollector
+	storageClient     StorageClient
+	fs                afero.Fs
+
+	dir string
+}
+
+func NewAggregator(resourceCollector ResourceCollector, logCollector LogCollector, storageClient StorageClient, fs afero.Fs, dir string) *Aggregator {
+	return &Aggregator{resourceCollector: resourceCollector, logCollector: logCollector, storageClient: storageClient, fs: fs, dir: dir}
+}
+
+func DefaultAggregator() (*Aggregator, error) {
+	fs := afero.NewOsFs()
+	storageClient := NewFileStorageClient(fs)
+	resourceCollector, err := DefaultResourceCollector()
+	if err != nil {
+		return nil, errors.InitializationError(err, aggregatorName)
+	}
+	logCollector, err := DefaultLogCollector()
+	if err != nil {
+		return nil, errors.InitializationError(err, aggregatorName)
+	}
+	tmpd, err := afero.TempDir(fs, "", "")
+	if err != nil {
+		return nil, err
+	}
+	return &Aggregator{
+		logCollector:      logCollector,
+		resourceCollector: resourceCollector,
+		fs:                fs,
+		dir:               tmpd,
+		storageClient:     storageClient,
+	}, nil
+
+}
+
+func (a *Aggregator) StreamFromManifest(manifest helmchart.Manifests, namespace, filename string) error {
+	if err := a.createSubResourceDirectories(); err != nil {
+		return err
+	}
+	unstructuredResources, err := manifest.ResourceList()
+	if err != nil {
+		return err
+	}
+	kubeResources, err := a.resourceCollector.RetrieveResources(unstructuredResources, namespace, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	if err := a.resourceCollector.SaveResources(filepath.Join(a.dir, "resources"), kubeResources); err != nil {
```

Comment:
pull "resources" and "logs" into consts, consider moving filepath.Join(...) into a private utility here.